File: 4-median-of-two-sorted-arrays/brute-force/brute-force.cpp
#include <vector>

class Solution
{
    public:
        double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2)
        {
            int m = nums1.size();
            int n = nums2.size();
            int t = m + n;

            vector<int> mergedArr(t);
            double median = 0.0;

            int i = 0, j = 0, k = 0;

            while ((i < m) && (j < n))
            {
                if (nums1[i] < nums2[j])
                {
                    mergedArr[k++] = nums1[i++];
                }
                else
                {
                    mergedArr[k++] = nums2[j++];
                }
            }

            while (i < m)
            {
                mergedArr[k++] = nums1[i++];
            }

            while (j < n)
            {
                mergedArr[k++] = nums2[j++];
            }

            if ((t % 2) == 0)
            {
                int a = mergedArr[t / 2];
                int b = mergedArr[(t / 2) - 1];

                median = (double)(a + b) / 2.0;
            }
            else
            {
                median = (double)(mergedArr[(m + n) / 2]);
            }

            return median;
        }
};

File: 0004-median-of-two-sorted-arrays/brute-force/brute-force.rs
impl Solution
{
    pub fn find_median_sorted_arrays(nums1: Vec<i32>, nums2: Vec<i32>) -> f64
    {
        let mut mergedVec: Vec<i32> = Vec::new();
        let mut median: f64 = 0.0;

        let mut i: usize = 0;
        let mut j: usize = 0;
        
        let m: usize = nums1.len() as usize;
        let n: usize = nums2.len() as usize;

        while ((i < m) && (j < n))
        {
            if (nums1[i] < nums2[j])
            {
                mergedVec.push(nums1[i]);
                i += 1;
            }
            else
            {
                mergedVec.push(nums2[j]);
                j += 1;
            }
        }

        while (i < m)
        {
            mergedVec.push(nums1[i]);
            i += 1;
        }

        while (j < n)
        {
            mergedVec.push(nums2[j]);
            j += 1;
        }

        let odd: u8 = ((nums1.len() + nums2.len()) % 2) as u8;

        if (odd == 0)
        {
            let idx1: usize = ((nums1.len() + nums2.len()) / 2) as usize;
            let idx2: usize = (((nums1.len() + nums2.len()) / 2) - 1) as usize;

            let a: i32 = mergedVec[idx1];
            let b: i32 = mergedVec[idx2];
            let tmp: f64 = (a + b) as f64;

            median = tmp / 2.0;
        }
        else
        {
            let idx: usize = ((nums1.len() + nums2.len()) / 2) as usize;
            median = mergedVec[idx] as f64;
        }

        return median
    }
}